src/remote/inet.cpp

```diff
@@ -1486,10 +1486,11 @@ static rem_port* aux_connect(rem_port* port, PACKET* packet)
 		port->auxAcceptError(packet);
 		inet_error(false, port, "socket", isc_net_event_connect_err, savedError);
 	}
+	new_port->port_handle = n;
 
```

Comment:
It seems not fully correct to assign not connected socket here.

Reply:
Why not? n has the new port handle here. In the previous version, n was assigned to new_port->port_handle just a few lines below, I just moved it up so the setFastLoopbackOption() method can already use the port_handle value. If the method runs into an exception e.g. on connect, the new_port object won't be returned to the caller anyway.

Reply:
A lot of reasons: in case of error, `n` is closed but `port_handle` will still point to the incorrect socket. Another thread could allocate new socket and it could get same value. Also, `new_port` is still linked with parent port.

Reply:
I see your point.